repose-aggregator/components/filters/header-translation-filter/src/main/scala/org/openrepose/filters/headertranslation/HeaderTranslationFilter.scala

```diff
@@ -74,10 +74,19 @@ class HeaderTranslationFilter @Inject()(configurationService: ConfigurationServi
         if (originalHeaderValues.nonEmpty) {
           sourceHeader.getNewName foreach { newHeaderName =>
             originalHeaderValues foreach { originalHeaderValue =>
-              if (quality.isEmpty) {
-                httpRequest.addHeader(newHeaderName, originalHeaderValue)
+              def writeHeaders(writeWithoutQuality: (String, String) => Unit,
+                               writeWithQuality: (String, String, Double) => Unit): Unit = {
+                if (quality.isEmpty) {
+                  writeWithoutQuality(newHeaderName, originalHeaderValue)
+                } else {
+                  writeWithQuality(newHeaderName, withoutQuality(originalHeaderValue), quality.get)
+                }
+              }
```

Comment:
That if-else is pretty nasty. The other option is to nest an if that is identical inside both sides of the if else, which i didn't do because i didn't want the code duplication. This way shows the originals intention of doing it with or without the quality, but allows me to make the distinction between adding and replacing once and in a simple manner.